code/modules/xenomorph/xeno_structures.dm

```diff
@@ -452,3 +452,217 @@
 	newshot.permutated += src
 	newshot.def_zone = pick(GLOB.base_miss_chance)
 	newshot.fire_at(hostile, src, null, ammo.max_range, ammo.shell_speed)
+
+
+//////////////////////
+///// xeno jell turret
+//////////////////////
+
+/obj/structure/xeno/resin/xeno_turret_jel
```

Comment:
Same thing here as with `/datum/action/xeno_action/activable/build_turret_jel`. By not making it a subtype you're having to redefine a *lot* of code.
If this was `/obj/structure/xeno/resin/xeno_turret/jel` or something you'd only need to change a few things for it to work.